### PR TITLE
feature: add options command for show global flag

### LINKED ISSUE #8147: Global flags no longer shown
I'm opening this issue to make sure this get properly tracked and fixed before the next release.

With #7917, global flags are no longer being shown in the main help text (running `helm` or `helm -h`).  This was first reported here https://github.com/helm/helm/pull/7917#issuecomment-626189561
This may be a good thing, as long as there is a way to see the global flags, which there isn't right now. 

Furthermore, the help text of each sub-command refers to a non-existant `helm <sub-command> options`.

For example:
```
$ helm status -h
[...]
Usage:
  helm status RELEASE_NAME [flags] [options]

Use "helm status options" for a list of global command-line options (applies to all commands).
```
but `helm status options` does not exist.

PR #8095 aims to fix that.

Output of `helm version`:
version.BuildInfo{Version:"v3.2+unreleased", GitCommit:"3779d959667095e03308a2e49480748f7911b228", GitTreeState:"dirty", GoVersion:"go1.13.7"}

### PR DESCRIPTION
**Pull request by zgfh**
**What this PR does / why we need it**:
add options command for show global flag
fix  https://github.com/helm/helm/pull/7917#issuecomment-626189561
**Special notes for your reviewer**:

**If applicable**:
- [x] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

File: cmd/helm/options.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	"helm.sh/helm/v3/cmd/helm/require"
+	"io"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+// NewCmdOptions implements the options command
+func NewCmdOptions(out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "options",
+		Short: "Print the list of flags inherited by all commands",
```
Comment (liuming-dev) [inline cmd/helm/options.go:30]: first word in messge of `Short` should be lower camel case, see #8054

Comment (zgfh) [reply cmd/helm/options.go]: fixed ,thanks

**Review by jdolitsky — APPROVED** [review]
This looks good. Could you add the documentation on this new flag in [helm-www](https://github.com/helm/helm-www)?

File: cmd/helm/root.go [hunk 2]
```diff
@@ -74,6 +74,7 @@ func newRootCmd(actionConfig *action.Configuration, out io.Writer, args []string
 		Use:                    "helm",
 		Short:                  "The Helm package manager for Kubernetes.",
 		Long:                   globalUsage,
+		Run:                    runHelp,
```
Comment (marckhouzam) [inline cmd/helm/root.go:77]: Could you in-line this as you did for the `options` command?
Then can you put a comment to why we need to specify the `Run` field?
I'm worried that someone could remove it during a refactor and wouldn't notice the impact.

File: cmd/helm/options.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+		Use:   "options",
+		Short: "print the list of flags inherited by all commands",
+		Long:  "print the list of flags inherited by all commands",
+		Args:  require.NoArgs,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmd.Usage()
+		},
+	}
+
+	// The `options` command needs write its output to the `out` stream
```
Comment (marckhouzam) [inline cmd/helm/options.go:38]: Missing "to" in "needs to write its output"

**Review by marckhouzam — CHANGES_REQUESTED** [review]
There is another PR that was planning on removing the many klogs flags. See #8067.

We need to sort out what we want to do between this approach, #8067, or both.

**Comment by marckhouzam** [comment]
Another problem this PR fixes is that with #7917, the help text of each sub-command refers to a non-existant `helm <sub-command> options`.

For example:
```
$ helm status -h
[...]
Usage:
  helm status RELEASE_NAME [flags] [options]

Use "helm status options" for a list of global command-line options (applies to all commands).
```
but `helm status options` does not exist.

**Comment by zgfh** [comment]
> Another problem this PR fixes is that with #7917, the help text of each sub-command refers to a non-existant `helm <sub-command> options`.
> 
> For example:
> 
> ```
> $ helm status -h
> [...]
> Usage:
>   helm status RELEASE_NAME [flags] [options]
> 
> Use "helm status options" for a list of global command-line options (applies to all commands).
> ```
> 
> but `helm status options` does not exist.

let me try

File: cmd/helm/options.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+import (
+	"github.com/spf13/cobra"
+	"helm.sh/helm/v3/cmd/helm/require"
+	"io"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+// NewCmdOptions implements the options command
+func NewCmdOptions(out io.Writer) *cobra.Command {
```
[the remark this answers is not in the document]
Comment (zgfh) [reply cmd/helm/options.go]: thanks ,fixed

**Comment by zgfh** [comment]
@marckhouzam fix #8149 at https://github.com/helm/helm/pull/8095/commits/c2069f7e8b95d6817878ef1444864aafb1d3c6c4

**Pull request closed without merging**